### PR TITLE
[5.7] Add boundaries() method to Collection.

### PR DESCRIPTION
**Pull request by lucasmichot**
Add `boundaries()` method to `\Illuminate\Support\Collection`.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -1125,6 +1126,20 @@ public function min($callback = null)
         });
     }
 
+    /**
+     * Get the boundaries for a given key.
+     *
+     * @param  callable|string|null  $callback
+     * @return mixed
```
Comment (ntzm) [inline src/Illuminate/Support/Collection.php]:
```suggestion
     * @return array
```

**Comment by deleugpn** [comment]
Shouldn't it return an instance of collection instead? Every time I call a collection method I kind of expect another collection back

**Comment by lucasmichot** [comment]
The purpose of this PR is really to provide boundaries.
I see your point @deleugpn , but I would not expect `min`, `max`, `average` or `mode` to return an instance.
The same applies to this method

**Comment by DojoGeekRA** [comment]
@lucasmichot 

`min`, `max` and `average` by concept return just one element of the collection, but this is not the case.

If i wanted to do something with each element of the `boundaries` result, i would expect something like this to be possible:

```php
$collection->boundaries()->each(function ($element) {
    foo($element);
});
```

**Comment by lucasmichot** [comment]
I see your point
@taylorotwell let me know your thoughts

**Comment by taylorotwell** [comment]
Hmm, I would prefer to see this macro'd in maybe. I'm not sure it has a wide use case.

**Pull request closed without merging**